### PR TITLE
Implement design

### PR DESCRIPTION
**Pull request by BakkerTom**
In this Pull Request:
- I implemented the graphical design as described in the UI mockups
- Deciding on what kind of card to use based on the availability of a header image (Not tested)
- Lazy loading of images with Ion
- Neat formatting of Timestamps relative to current time (eg: '4 hours ago')

---
Before submitting your pull request, consider the following: Does your code...

- [x] Implement functionality
- [x] Contain comments/documentation
- [x] Build without errors
- [x] Contain and pass unit testing
- [x] Conform to the Style Guide
- [x] Have no merge conflicts with ```master```

When you've checked all of the boxes above your code is up for review. If approved it can be merged with the ```master``` branch.

If you want feedback on your work before having finished your code, add ```[WIP]``` to the title of your PR.

### CODE REVIEW

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java [hunk 1]
```diff
@@ -26,22 +32,53 @@ public PostAdapter(Context context, @LayoutRes int resource) {
...
-            xml_type = R.layout.list_item_post_2;
+            xmlType = R.layout.list_item_post_image;
         }
+
         // Check if an existing view is being reused, otherwise inflate the view
         if (convertView == null) {
-            convertView = LayoutInflater.from(getContext()).inflate(xml_type, parent, false);
-            TextView postName = (TextView) convertView.findViewById(R.id.postTitle);
-            postName.setText(p.getTitle());
+            convertView = LayoutInflater.from(getContext()).inflate(xmlType, parent, false);
```
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: What happens if `convertView` is not null, and is of the wrong type? So, say it has been created as a view without an image, it then gets reused for a view *with* an image. I'm not sure if our code can handle this.

File: android/build.gradle [hunk 2]
```diff
@@ -37,4 +37,5 @@ dependencies {
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
+    compile 'com.android.support:cardview-v7:25.2.0'
```
Comment (SanderVanAndel) [inline android/build.gradle]: Think this will look more neat if the line is above the androidTestCompile

**Comment by SvenDub** [comment]
Love the design. Would it be possible to get the ripple effect on the card when you touch it instead of behind the card?

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java (hunk 1 above)
Comment (BakkerTom) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: That was exactly what I noticed last night, and what I wanted to discuss with you guys.

Do you think we should solve this issue within the current sprint, or push it over to the next sprint when we introduce more types of content?

File: android/build.gradle (hunk 2 above)
Comment (BakkerTom) [inline android/build.gradle]: No problemo

File: android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java (hunk 1 above)
Comment (SvenDub) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: Looking at the current dataset, it is possible for almost every field returned by the api to be `null`. So even now it is possible to receive a post without an image. I think we should fix that now.

Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/Adapters/PostAdapter.java]: I ran the code with a small change so cards with and without an image get used, this didn't cause any errors

**Pull request merged**